src/server/pps/cmds/defaults_test.go

```diff
@@ -0,0 +1,250 @@
+				"cpu": null,
+				"disk": "187Mi"
+			},
+			"autoscaling": false
+		}
+		EOF
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .details.resource_requests.disk | match 187Mi
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .details.resource_requests.memory | match 64M
+		`,
+		"ProjectName", pfs.DefaultProjectName,
+		"RepoName", "input",
+		"PipelineName", "pipeline",
+	).Run())
+}
+
+func TestCreatePipeline_defaults(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c, _ := minikubetestenv.AcquireCluster(t)
+	require.NoError(t, tu.PachctlBashCmd(t, c, `
+		pachctl create defaults --cluster <<EOF
+		{
+			"create_pipeline_request": {
+				"datumTries": 17,
+				"autoscaling": true
+			}
+		}
+		EOF
+		pachctl create repo {{.RepoName}}
+		pachctl create pipeline <<EOF
+		{
+			"pipeline": {
+				"project": {
+					"name": "{{.ProjectName | js}}"
+				},
+				"name": "{{.PipelineName | js}}"
+			},
+			"transform": {
+				"cmd": ["cp", "r", "/pfs/in", "/pfs/out"]
+			},
+			"input": {
+				"pfs": {
+					"project": "default",
+					"repo": "{{.RepoName | js}}",
+					"glob": "/*",
+					"name": "in"
+				}
+			},
+			"resource_requests": {
+				"cpu": null,
+				"disk": "187Mi"
+			},
+			"autoscaling": false
+		}
+		EOF
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .details.resource_requests.disk | match 187Mi
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .details.resource_requests.memory | match null
+		# the raw format currently marshals to snake case
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .details.datum_tries | match 17
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .details.autoscaling | match null
+		# the specs marshal to camelCase
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .effective_spec_json | jq -r .datumTries | match 17
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .effective_spec_json | jq -r .autoscaling | match false
+		`,
+		"ProjectName", pfs.DefaultProjectName,
+		"RepoName", "input",
+		"PipelineName", "pipeline",
+	).Run())
+}
+
+func TestCreatePipeline_regenerate(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c, _ := minikubetestenv.AcquireCluster(t)
+	require.NoError(t, tu.PachctlBashCmd(t, c, `
+		pachctl create defaults --cluster <<EOF
+		{
+			"create_pipeline_request": {
+				"datumTries": 17,
+				"autoscaling": true
+			}
+		}
+		EOF
+		pachctl create repo {{.RepoName}}
+		pachctl create pipeline <<EOF
+		{
+			"pipeline": {
+				"project": {
+					"name": "{{.ProjectName | js}}"
+				},
+				"name": "{{.PipelineName | js}}"
+			},
+			"transform": {
+				"cmd": ["cp", "r", "/pfs/in", "/pfs/out"]
+			},
+			"input": {
+				"pfs": {
+					"project": "default",
+					"repo": "{{.RepoName | js}}",
+					"glob": "/*",
+					"name": "in"
+				}
+			},
+			"resource_requests": {
+				"cpu": null,
```

Comment:
why do we set cpu to null?

Reply:
Copy-pasted from the API lifecycle tests. But I’ll add an explicit check for the expected results.